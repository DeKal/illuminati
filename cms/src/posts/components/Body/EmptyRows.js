import React from 'react'
import PropTypes from 'prop-types'
import TableCell from '@material-ui/core/TableCell'
import TableRow from '@material-ui/core/TableRow'
import { DEFAULT_LINE_HEIGHT } from 'core/const/posts'

const EmptyRows = ({ page, rowsPerPage, postsLength }) => {
  const emptyRows =
    rowsPerPage - Math.min(rowsPerPage, postsLength - page * rowsPerPage)
  return emptyRows ? (
    <TableRow style={{ height: DEFAULT_LINE_HEIGHT * emptyRows }}>
      <TableCell colSpan={6} />
    </TableRow>
  ) : null
}

export default EmptyRows

EmptyRows.propTypes = {
  page: PropTypes.number,
  rowsPerPage: PropTypes.number,
  postsLength: PropTypes.number
}
