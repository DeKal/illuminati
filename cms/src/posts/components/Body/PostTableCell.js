import { withStyles } from '@material-ui/core/styles'
import TableCell from '@material-ui/core/TableCell'

const PostTableCell = withStyles(theme => ({
  root: {
    width: '50%',
    cursor: 'pointer',
    whiteSpace: 'nowrap',
    '&:hover': {
      color: theme.colors.primary
    }
  }
}))(TableCell)

export default PostTableCell
