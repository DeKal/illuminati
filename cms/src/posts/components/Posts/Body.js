import React from 'react'
import PropTypes from 'prop-types'
import TableBody from '@material-ui/core/TableBody'
import TableCell from '@material-ui/core/TableCell'
import TableRow from '@material-ui/core/TableRow'
import Checkbox from '@material-ui/core/Checkbox'
import { stableSort, getComparator } from 'core/utils/sort'

const Body = ({
  posts,
  page,
  rowsPerPage,
  order,
  selectedPosts,
  orderColumn,
  setSelectedPost
}) => (
  <TableBody>
    {stableSort(posts, getComparator(order, orderColumn))
      .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
      .map((row, index) => {
        const isItemSelected = selectedPosts.indexOf(row.name) !== -1
        const labelId = `enhanced-table-checkbox-${index}`

        return (
          <TableRow
            hover
            onClick={() => setSelectedPost(row.name)}
            role="checkbox"
            aria-checked={isItemSelected}
            tabIndex={-1}
            key={row.name}
            selected={isItemSelected}
          >
            <TableCell padding="checkbox">
              <Checkbox
                checked={isItemSelected}
                inputProps={{ 'aria-labelledby': labelId }}
              />
            </TableCell>
            <TableCell component="th" id={labelId} scope="row" padding="none">
              {row.name}
            </TableCell>
            <TableCell align="right">{row.calories}</TableCell>
            <TableCell align="right">{row.fat}</TableCell>
            <TableCell align="right">{row.carbs}</TableCell>
            <TableCell align="right">{row.protein}</TableCell>
          </TableRow>
        )
      })}
    {renderEmptyRows(page, rowsPerPage, posts.length)}
  </TableBody>
)

export default Body

Body.propTypes = {
  posts: PropTypes.array,
  page: PropTypes.number,
  rowsPerPage: PropTypes.number,
  order: PropTypes.string,
  orderColumn: PropTypes.string,
  selectedPosts: PropTypes.array,
  setSelectedPost: PropTypes.func
}

const renderEmptyRows = (page, rowsPerPage, postLength) => {
  const emptyRows =
    rowsPerPage - Math.min(rowsPerPage, postLength - page * rowsPerPage)
  return emptyRows ? (
    <TableRow style={{ height: 53 * emptyRows }}>
      <TableCell colSpan={6} />
    </TableRow>
  ) : null
}
