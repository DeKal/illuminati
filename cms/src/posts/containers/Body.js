import { connect } from 'react-redux'
import { withRouter } from 'react-router'
import Body from 'posts/components/Posts/Body'
import { setSelectedPost } from 'posts/state/actions'
import {
  getPage,
  getRowsPerPage,
  getOrder,
  getOrderColumn
} from 'posts/state/selectors'

const mapStateToProps = state => ({
  page: getPage(state),
  rowsPerPage: getRowsPerPage(state),
  order: getOrder(state),
  orderColumn: getOrderColumn(state)
})

const mapDispatchToProps = {
  setSelectedPost
}

const BodyWithRouter = withRouter(Body)
export default connect(mapStateToProps, mapDispatchToProps)(BodyWithRouter)
