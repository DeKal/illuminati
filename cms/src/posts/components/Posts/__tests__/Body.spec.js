import * as React from 'react'
import { shallow } from 'enzyme'
import Body from '../Body'

const posts = [
  {
    slug: 'getting-started-with-java',
    title: 'Getting Started With Java',
    time: '06/01/2020',
    tag: ['Web Development', 'Java'],
    author: 'Phat Ho',
    commentNum: 0
  },
  {
    slug: 'get-startted-with-spring',
    title: 'Get Startted With Spring MVC',
    time: '06/01/2020',
    tag: ['Java Spring', 'Course'],
    author: 'Phat Ho',
    commentNum: 0
  }
]

describe('Body', () => {
  it('should render with no error', function() {
    const wrap = shallow(
      <Body
        posts={posts}
        page={0}
        rowsPerPage={5}
        order={'asc'}
        orderColumn={'title'}
        selectedPosts={[]}
      />
    )
    expect(wrap).toMatchSnapshot()
  })

  it('should change page when click on title', function() {
    const history = {
      push: jest.fn()
    }
    const wrap = shallow(
      <Body
        posts={posts}
        page={0}
        rowsPerPage={5}
        order={'asc'}
        orderColumn={'title'}
        selectedPosts={[]}
        history={history}
      />
    )
    wrap.find('[data-test-id="body-title-0"]').simulate('click')
    expect(history.push.mock.calls.length).toBe(1)
  })

  it('should check the item when clicking on checkbox', function() {
    const setSelectedPost = jest.fn()
    const wrap = shallow(
      <Body
        posts={posts}
        page={0}
        rowsPerPage={5}
        order={'asc'}
        orderColumn={'title'}
        selectedPosts={[]}
        setSelectedPost={setSelectedPost}
      />
    )
    wrap.find('[data-test-id="body-checkbox-0"]').simulate('click')
    expect(setSelectedPost.mock.calls.length).toBe(1)
    expect(setSelectedPost.mock.calls[0][0]).toEqual('get-startted-with-spring')
  })
})
