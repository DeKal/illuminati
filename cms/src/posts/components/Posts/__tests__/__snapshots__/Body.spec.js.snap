// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body should render with no error 1`] = `
<WithStyles(ForwardRef(TableBody))>
  <WithStyles(ForwardRef(TableRow))
    aria-checked={false}
    hover={true}
    key="get-startted-with-spring"
    role="checkbox"
    selected={false}
    tabIndex={-1}
  >
    <WithStyles(ForwardRef(TableCell))
      padding="checkbox"
    >
      <WithStyles(ForwardRef(Checkbox))
        checked={false}
        data-test-id="body-checkbox-0"
        inputProps={
          Object {
            "aria-labelledby": "enhanced-table-checkbox-0",
          }
        }
        onClick={[Function]}
      />
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(WithStyles(ForwardRef(TableCell)))
      component="th"
      data-test-id="body-title-0"
      id="enhanced-table-checkbox-0"
      onClick={[Function]}
      padding="none"
      scope="row"
    >
      Get Startted With Spring MVC
    </WithStyles(WithStyles(ForwardRef(TableCell)))>
    <WithStyles(ForwardRef(TableCell))>
      Phat Ho
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(TableCell))>
      06/01/2020
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(TableCell))
      style={
        Object {
          "whiteSpace": "nowrap",
        }
      }
    >
      Java Spring, Course
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(TableCell))
      align="right"
    >
      0
    </WithStyles(ForwardRef(TableCell))>
  </WithStyles(ForwardRef(TableRow))>
  <WithStyles(ForwardRef(TableRow))
    aria-checked={false}
    hover={true}
    key="getting-started-with-java"
    role="checkbox"
    selected={false}
    tabIndex={-1}
  >
    <WithStyles(ForwardRef(TableCell))
      padding="checkbox"
    >
      <WithStyles(ForwardRef(Checkbox))
        checked={false}
        data-test-id="body-checkbox-1"
        inputProps={
          Object {
            "aria-labelledby": "enhanced-table-checkbox-1",
          }
        }
        onClick={[Function]}
      />
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(WithStyles(ForwardRef(TableCell)))
      component="th"
      data-test-id="body-title-1"
      id="enhanced-table-checkbox-1"
      onClick={[Function]}
      padding="none"
      scope="row"
    >
      Getting Started With Java
    </WithStyles(WithStyles(ForwardRef(TableCell)))>
    <WithStyles(ForwardRef(TableCell))>
      Phat Ho
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(TableCell))>
      06/01/2020
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(TableCell))
      style={
        Object {
          "whiteSpace": "nowrap",
        }
      }
    >
      Web Development, Jav...
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(TableCell))
      align="right"
    >
      0
    </WithStyles(ForwardRef(TableCell))>
  </WithStyles(ForwardRef(TableRow))>
  <EmptyRows
    page={0}
    postsLength={2}
    rowsPerPage={5}
  />
</WithStyles(ForwardRef(TableBody))>
`;
